src/modules/commander/Commander.cpp

```diff
@@ -1984,6 +1987,47 @@ void Commander::safetyButtonUpdate()
 	}
 }
 
+void Commander::throwLaunchUpdate()
+{
+	if (_param_com_throw_en.get()) {
+		if (_vehicle_local_position_sub.updated()) {
+			_vehicle_local_position_sub.copy(&_vehicle_local_position);
+		}
+
+		if (!isArmed() && _throw_launch_state != ThrowLaunchState::IDLE) {
+			mavlink_log_info(&_mavlink_log_pub, "The vehicle is DISARMED with throw launch enabled. Do NOT throw it.\t");
+			_throw_launch_state = ThrowLaunchState::IDLE;
+		}
+
+		if (_throw_launch_state == ThrowLaunchState::IDLE && isArmed()) {
```

Comment:
A bit of nitpicking here, but a `switch` structure could highlight the state machine a bit better:
```
switch (_throw_launch_state) {
case ThrowLaunchState::IDLE:
	if (isArmed()) {
		mavlink_log_info(&_mavlink_log_pub, "The vehicle is ARMED with throw launch enabled. Throw the vehicle now.\t");
		_throw_launch_state = ThrowLaunchState::ARMED;
	}
	break;
case ThrowLaunchState::ARMED:
....
```

Reply:
Good point, indeed looks better now!